acme_test.go

```diff
@@ -155,6 +159,81 @@ func TestAuthorize(t *testing.T) {
 	}
 }
 
+func TestAcceptChallenge(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == "HEAD" {
+			w.Header().Set("replay-nonce", "test-nonce")
+			return
+		}
+		if r.Method != "POST" {
+			t.Errorf("r.Method = %q; want POST", r.Method)
+		}
+
+		// Decode request
+		body, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			t.Fatal(err)
+		}
+		var bodyJSON map[string]string
+		err = json.Unmarshal(body, &bodyJSON)
+		payload, err := base64.RawURLEncoding.DecodeString(bodyJSON["payload"])
+		if err != nil {
+			t.Fatal(err)
+		}
+		var j map[string]string
+		err = json.Unmarshal(payload, &j)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Test request
+		if j["resource"] != "challenge" {
+			t.Errorf(`resource = %q; want "challenge"`, j["resource"])
+		}
+		if j["type"] != "http-01" {
+			t.Errorf(`type = %q; want "http-01"`, j["type"])
+		}
+		if !strings.HasPrefix(j["keyAuthorization"], "token1.") {
+			t.Errorf(`keyAuthorization = %q; want "token1.[...]"`, j["keyAuthorization"])
+		}
+
+		// Respond to request
+		w.WriteHeader(http.StatusAccepted)
+		fmt.Fprintf(w, `{
+			"type":"http-01",
+			"status":"pending",
+			"uri":"https://ca.tld/acme/challenge/publickey/id1",
+			"token":"token1",
+			"keyAuthorization":"token.jwt"}`)
+	}))
+	defer ts.Close()
+	key, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		t.Fatal(err)
+	}
+	cfg := &Config{
+		Key: key,
+	}
+	c, err := acceptChallenge(nil, cfg, Challenge{
```

Comment:
I'd inline cfg, you're not using it anywhere else and it's quite small:

    acceptChallenge(nil, &Config{Key: key}, Challenge{